Rewrite conv testers from cpp to python

paulinagacek opened the pull request:
### PR types
Others

### PR changes
Others

### Describe
Deleted old C++ scale matmul fuse pass test and updated python one

Currently in PaddlePaddle there are 3 unit-test frameworks - one written in C++ and 2 in Python.
Old fuse pass tests (C++) fuse_pass_tester.cc are checking only if graph has correct number of / type of vertices after fusion. It doesn't check data. So these tests pass even if accuracy drops.
New python framework is superior because it checks both graph fusion and accuracy after fusion.

Tests rewritten in this PR:

- conv_bias_mkldnn_fuse_pass_tester.cc
- conv_concat_relu_mkldnn_fuse_pass_tester.cc
- conv_elementwise_add_mkldnn_fuse_pass_tester.cc

Commit:
conv_bias_mkldnn_fuse_pass_tester rewritten

Commit:
conv_concat_relu_mkldnn_fuse_pass_tester rewritten

Commit:
conv_elementwise_add_fuse_pass_tester rewritten

Commit:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into rewrite_conv_testers

Commit:
mkldnn changed to onednn

Commit:
tests added to cmakeLists, style fix

paulinagacek commented:
@Silv3S @HulekJakub can you review please?

HulekJakub commented on python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bias_fuse_pass.py:
```diff
@@ -22,8 +22,7 @@
 
-# padding SAME
-class ConvBiasMkldnnFusePassSamePadTest(InferencePassTest):
+class TestOneDNNConvBiasFusePassPaddingSame(InferencePassTest):
```
I think new pass tests should derive from PassAutoScanTest, you can use code from the deleted file'test_conv_bias_mkldnn_fuse_pass.py'

paulinagacek replied:
you are right, thanks

HulekJakub reviewed (dismissed):
LGTM

Commit:
got rid of unnecessary UT, some style changes

Commit:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into rewrite_conv_testers

paulinagacek commented:
@kolinwei @XieYunshen please review and approve 
**PR-CI-Coverage** (changes in unnitests). I deleted c++ tests becuase they were no longer needed (more about it in description above) and renamed python UT to match new convention.

Silv3S commented on python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_concat_relu_fuse_pass.py:
```diff
@@ -21,7 +21,7 @@
 
 
-class TestConvConcatActivationMkldnnFusePass(PassAutoScanTest):
+class TestOneDNNConvConcatActivationFusePass(PassAutoScanTest):
```
This file should be renamed to _test_onednn_conv_concat_activation_fuse_pass.py_ as it's no longer limited to `ReLU`.

Silv3S commented on python/paddle/fluid/tests/unittests/ir/inference/test_onednn_conv_bias_fuse_pass.py:
```diff
@@ -20,66 +20,42 @@
 
 
 class TestConvBiasMkldnnFusePass(PassAutoScanTest):
```
```suggestion
class TestConvBiasOneDNNFusePass(PassAutoScanTest):
```

Silv3S commented on python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt:
```diff
@@ -221,7 +221,10 @@ if(WITH_GPU AND TENSORRT_FOUND)
 
   if(WITH_MKLDNN)
-    set_tests_properties(test_mkldnn_conv_elementwise_add_fuse_pass
+    set_tests_properties(test_onednn_conv_bias_fuse_pass PROPERTIES TIMEOUT 120)
+    set_tests_properties(test_onednn_conv_elementwise_add_fuse_pass
+                         PROPERTIES TIMEOUT 120)
+    set_tests_properties(test_onednn_conv_concat_relu_fuse_pass
```
```suggestion
    set_tests_properties(test_onednn_conv_elementwise_add_fuse_pass
                         PROPERTIES TIMEOUT 120)
```
It's not required to set timeout if tests are passing without it

Commit:
changes in naming convention

Commit:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into rewrite_conv_testers

Silv3S reviewed (dismissed):
LGTM

yaomichael commented:
@yeliang2258 can you help find someone to approve this PR?

qingqing01 reviewed (dismissed):
LGTM

XieYunshen reviewed (dismissed):
LGTM

Commit:
max_examples reduced

Commit:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into rewrite_conv_testers

zhwesky2010 reviewed (dismissed):
LGTM for change UT

paulinagacek commented:
@kolinwei @XieYunshen please approve PR-CI-Coverage and merge if everything is ok.

qingqing01 commented:
@PaulinaGacek still Timeout,  you can set TIMEOUT time in:

https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt#L231

Commit:
time out added

Commit:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into rewrite_conv_testers

paulinagacek commented:
@qingqing01 I've set time out, please check if it is ok now
